lib/compress/zstd_compress.c

```diff
@@ -1261,6 +1266,20 @@ MEM_STATIC size_t ZSTD_compressSequences (seqStore_t* seqStorePtr,
     { int i; for (i=0; i<ZSTD_REP_NUM; i++) seqStorePtr->rep[i] = seqStorePtr->repToConfirm[i]; }
 
     return op - ostart;
+
+_dst_size_too_small:
```

Comment:
My understanding is that this part is always triggered by ` ERROR(dstSize_tooSmall)`.

Therefore, could we nest it instead ?
`ZSTD_compressSequences()` would become `ZSTD_compressSequences_internal()` (for example),
it would be called by a new `ZSTD_compressSequences()`
which only role is to read the return code, and trigger raw uncompressed block when it is `== ERROR(dstSize_tooSmall)`.

It's basically the same, but feels cleaner than a `goto` statement.
What do you think ?